### PR TITLE
Add DESCRIPTION checks for CRAN formatting and best practices

### LINKED ISSUE #85: Check for package and function names in DESCRIPTION
CRAN requires that package names be single-quotes and function names have parentheses but this isn't covered by R CMD check.  We could test for this in **goodpractice**.  First, we could check if any of the packages in Imports/Depends/Suggests/Enhances is in these fields.  Secondly, we could look for exported functions for the package itself.  Looking for package names and functions across all packages probably doesn't make sense, as the possible names would be too large.

### LINKED ISSUE #122: Some ideas for DESCRIPTION checking
Possibly some of these are already implemented 🙈 

* valid URLs (`crul::ok()` or `httr` equivalent) anywhere where there's an URL - including checking whether there's a redirect, i.e. when a repo was transferred to a new owner on GitHub, the URL should be changed although the old one is still valid because redirected.
* has the URLs (BugReports, Repository)
* only roles that exist
* no duplicate authors (tricky one when not for exact match)
* the description check that's already done somewhere else (stating it shouldn't start by "This package")
* no duplicate dependency, e.g. the same package twice once under Depends once under Suggests
* no missing necessary field
* no non-accepted fields
* ORCIDs (_that_ would be "best-practice" and quite opiniated so probably not in scope) also cf ropensci-review-tools/pkgcheck#200 
* Anything that's in `R CMD Check` reg DESCRIPTION except with regard to the rest of the package.

### PR DESCRIPTION
**Pull request by drmowinckels**
## Summary

- New check: Description field should not start with "This package"
- New check: URLs in Description must be enclosed in angle brackets (`<https://...>`)
- New check: DOIs should use `<doi:10.prefix/suffix>` format, not full URLs
- New check: URLs should use `https://` not `http://`
- New check: no duplicate packages across dependency fields (e.g. same package in both Imports and Suggests)
- New check: Authors@R roles must be valid MARC relator codes
- New check: package names mentioned in Title/Description must be single-quoted

All checks use the existing `description` prep — no new dependencies or prep steps needed.

Closes #122, closes #85

## Test plan

- [x] All 151 tests pass
- [x] Each check tested with passing and failing DESCRIPTION fixtures
- [x] try-error early returns tested for all checks
- [x] Edge cases: broken Authors@R returns NA, no-deps returns TRUE for quoting check
- [ ] `R CMD check`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: R/chk_description.R [hunk 1]
```diff
@@ -61,6 +61,175 @@ CHECKS$description_url <- make_check(
...
+## --------------------------------------------------------------------
+
+CHECKS$description_urls_not_http <- make_check(
+
+  description = "URLs in Description use https not http",
+  tags = c("info", "DESCRIPTION"),
+  preps = "description",
+
+  gp = 'use https:// instead of http:// for URLs in the Description
+        field. CRAN prefers secure URLs where available.',
```
Comment (mpadge) [inline R/chk_description.R]: This is the only one I'm not sure about. "where available" is they key - secure ones may often not be available. And global distributions of https vs http are [uneven](https://pulse.internetsociety.org/en/technologies/?trk=article-ssr-frontend-pulse_little-text-block), making https enforcement potentially discriminatory, which I would personally not be comfortable with at present.

Comment (drmowinckels) [reply R/chk_description.R]: how about we change the wording?  

> CRAN prefers secure URLs where available. Check if there might be a secure https:// instead of http:// for URLs in the description.

Comment (drmowinckels) [reply R/chk_description.R]: I dont know. Since CRAN does check this, and might comment on it, it would be nice to make people aware and help them be preemptive about it? I.e. if no https alternative exists, mention it in CRAN comments?

**Review by mpadge — APPROVED** [review]
